Rocket.Chat/Managers/Socket/Response/SocketHandlers.swift

```diff
@@ -13,71 +13,70 @@ import SwiftyJSON
 import Bugsnag
 
 extension SocketManager {
-    
+
     func handleMessage(_ response: JSON, socket: WebSocket) {
-        let result = SocketResponse(response, socket: socket)!
-        
+        guard let result = SocketResponse(response, socket: socket) else { return }
+
         guard result.msg != nil else {
             return Log.debug("Msg is invalid: \(result.result)")
         }
-        
+
         guard !result.isError() else {
             return handleError(result, socket: socket)
         }
-        
-        switch result.msg! {
-        case .Connected: return handleConnectionMessage(result, socket: socket)
-        case .Ping: return handlePingMessage(result, socket: socket)
-        case .Changed, .Added, .Removed: return handleModelUpdates(result, socket: socket)
-        case .Error, .Updated, .Unknown: break
+
+        switch result.msg {
+            case .some(.Connected): return handleConnectionMessage(result, socket: socket)
+            case .some(.Ping): return handlePingMessage(result, socket: socket)
+            case .some(.Changed), .some(.Added), .some(.Removed): return handleModelUpdates(result, socket: socket)
+            case .some(.Error), .some(.Updated), .some(.Unknown): fallthrough
```

Comment:
`fallthrough` will prevent us from calling the callback with the error, right? Maybe a `break` is better for now, since we only have one callback.